server/public/plugin/api.go

```diff
@@ -1411,6 +1412,30 @@ type API interface {
 	// @tag Group
 	// Minimum server version: 10.9
 	DeleteGroupConstrainedMemberships() *model.AppError
+
+	// MakeAuditRecord creates a new audit record from the provided information.
+	//
+	// @tag Audit
+	// Minimum server version: 10.10
+	MakeAuditRecord(event string, initialStatus string) *model.AuditRecord
+
+	// MakeAuditRecord creates a new audit record from the provided information.
+	//
+	// @tag Audit
+	// Minimum server version: 10.10
+	MakeAuditRecordWithRequest(event string, initialStatus string, ctx Context, userId, apiPath string) *model.AuditRecord
```

Comment:
Drive by review :car: Do these have to be RPC API methods? They don't seem to do anything server-specific? Can they just be functions in `model`?

Reply:
@hanzei you bring up a good point. I wanted this isolated from other definitions to make it clear this is plugin specific, but it doesn't need to go over RPC. I've moved them into it's own file in the `public/plugin` package, and we'll call it directly.